Add additional information about custom server config to custom-dns-config docs. (#7559)
gardener/gardener: docs/usage/custom-dns-config.md

         whoami
```

The port number 8053 in `global:8053` is the specific port that CoreDNS is bound to and cannot be changed to any other port if it should act on ordinary name resolution requests from pods. Otherwise, CoreDNS will open a second port, but you are responsible to direct the traffic to this port. `kube-dns` service in `kube-system` namespace will direct name resolution requests within the cluster to port 8053 on the CoreDNS pods.

> Moreover, additional network policies are needed to allow corresponding ingress traffic to CoreDNS pods

Reviewer: ```suggestion
Moreover, additional network policies are needed to allow corresponding ingress traffic to CoreDNS pods.
```

         whoami
```

> The port number 8053 in `global:8053` is the specific port that CoreDN is bound to and cannot be changed to any other port. 

Reviewer 2: ```suggestion
The port number 8053 in `global:8053` is the specific port that CoreDNS is bound to and cannot be changed to any other port. 
```

> The port number 8053 in `global:8053` is the specific port that CoreDN is bound to and cannot be changed to any other port. 

Reviewer: ```suggestion
The port number 8053 in `global:8053` is the specific port that CoreDNS is bound to and cannot be changed to any other port if it should act on ordinary name resolution requests from pods. Otherwise, CoreDNS will open a second port, but you are responsible to direct the traffic to this port. `kube-dns` service in `kube-system` namespace will direct name resolution requests within the cluster to port 8053 on the CoreDNS pods.
```

> In order for the destination DNS server to be reachable, it must listen on port 53 as it is required by network policies.

Reviewer: ```suggestion
In order for the destination DNS server to be reachable, it must listen on port 53 as it is required by network policies. Other ports are only possible if additional network policies allow corresponding egress traffic from CoreDNS pods.
```
